### DISCUSSION TITLE
The client hangs for a while after successful connection and streaming

### DISCUSSION THREAD

**Discussion by enseitankado** [reactphp/reactphp#580] [Q&A]
I am not sure if this is the feature I am looking for. I want to take precautions against the connection hanging situation to make my application more agile. My scenario is as follows: the client successfully connects to the server within the timeout period and the data communication is healthy. After a while, for some reason (it may be a firewall or an error on the server, it does not matter) the data flow of my client stops. My client thinks that it is still connected and waits until the data continues to reach again. There is no problem with the connection timeout in this example, I just want to be able to check the situations where the server stops sending data for a certain period of time or my client cannot receive data for a certain period of time (hangs). Does the library have a feature for such a check? Can you help?

**Comment by WyriHaximus** [comment]
Does the client hang, or is there no data at all coming over the line? It might be connected perfectly fine with no data coming over. Are other parts of the same event loop working as expected? Which event loop are you using?

**Comment by PaulRotmann** [comment]
Since this post is a bit older and hasn't received a response yet, I assume this discussion can be considered closed at this point.

If you have any other questions come up in the future, feel free to start a new discussion thread.
